### PR TITLE
uses promises instead of callbacks

### PR DESCRIPTION
**Pull request by gonber**
BREAKING CHANGE

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,18 +19,17 @@ function Redis (host, port, pass, db = 0) {
   this.pass = pass;
   this.db = db;
   this.ready = false;
-  this.connect();
```
Comment (RaviPatel) [inline index.js]: curious: why not connect here?

File: index.js [hunk 2]
```diff
@@ -48,149 +49,139 @@ Redis.prototype.connect = function () {
...
-    callback(err, data);
-  });
+Redis.prototype.disconnect = async function () {
+  return this.client.quit();
 };
 
-Redis.prototype.get = function (key, callback) {
-  this.handle_client_oper_action('get', key, callback);
+Redis.prototype.handle_client_oper_action = async function (action, key) {
+  return tryExecuteAndLogError(this.client[action](key));
```
Comment (RaviPatel) [inline index.js]: minor: I prefer snake_case instead of camelCase

File: index.js (hunk 1 above)
Comment (gonber) [reply index.js]: the constructor cannot be async

Comment (RaviPatel) [reply index.js]: Aha, OK.. I'll need to take note of this when upgrading the repos .. do you think you can update READMEs for these packages with their use cases?

File: index.js (hunk 2 above)
Comment (gonber) [reply index.js]: applied

**Pull request merged**